### DISCUSSION TITLE
plotStationary question(s)

### DISCUSSION THREAD

**Discussion by rociojoo** [bmcclintock/momentuHMM#87] [Q&A] [answered]
Hi Brett, 
I'm wondering why you have !is.null(model$mod$hessian) as a condition to put  Sigma <- model$mod$Sigma in the plotStationary function (line 34 in plotStationary.momentuHMM (at least how I see it in RStudio). I guess it has to do with the way the covariance matrix of all parameters was estimated by numerical optimization.
Thanks in advance,
Rocío

**Comment by rociojoo** [comment]
Hi again,
I'd also like to know how and where you create the meansList object (line 53, same script). I can't find it anywhere. 
for (j in names(model$data)[which(unlist(lapply(model$data, 
      function(x) any(class(x) %in% meansList))))]) {
Thanks.

**Comment by rociojoo** [comment]
Also, the computed stationary distributions (and confidence intervals) are sensitive to the choice of values for the fixed covariates.

**Comment by bmcclintock** [comment, marked as answer]
Hi Rocio,

1) In order to plot confidence intervals, the hessian must have been calculated when fitting the model. If there is no hessian, then Sigma <- NULL and plotCI <- FALSE

2) This is defined in momentuHMM_utils.R.  This is for internal use and not exported, but you can view this object using the following: > momentuHMM:::meansList

3) I'm not sure what you mean by "sensitive" here, but I'll try and guess.  When covariates are included in the 'formula' argument, then the stationary distribution will depend on the covariate values.  The function plotStationary only plots one covariate at a time (if you want to plot multidimensional covariate plots then you're on your own :)), so when there are >1 covariates then the mean value of the other covariates (or the user-specified value in the 'covs' argument) is used.  See ?plotStationary and ?stationary.  If this doesn't address your comment, please feel free to clarify.

Cheers,
Brett

**Comment by rociojoo** [comment]
Hi Brett, thanks for your reply! I forgot to reply back. This answers all my questions, actually!
Cheers,
Rocío
